packages/next/src/client/components/app-router.tsx

```diff
@@ -376,14 +379,18 @@ function Router({
           })
         })
       },
-      replace: (href, options = {}) => {
+      replace: (untrustedHref, options = {}) => {
+        // Moving this check into the transition would trigger error boundaries
```

Comment:
You can call `router.push()` / `router.replace()` / `router.prefetch()` inside of a React transition, not sure it's intended that it would throw in that case?

I.e. this is valid:

```js
startTransition(() => {
  router.push('/')
  router.push('/dashboard')
})
```

Maybe it should `console.error` + ignore the navigation?

Reply:
True. `console.error` + ignore sounds like a good compromise. Doesn't surface as well though. I'll ask Seb since he did the original implementation in React if he has opinions about it.

Reply:
Done